field/src/packed.rs

```diff
@@ -146,27 +146,29 @@ unsafe impl<T: Packable, const WIDTH: usize> PackedValue for [T; WIDTH] {
     type Value = T;
     const WIDTH: usize = WIDTH;
 
+    #[inline]
     fn from_slice(slice: &[Self::Value]) -> &Self {
         assert_eq!(slice.len(), Self::WIDTH);
-        slice.try_into().unwrap()
+        unsafe { &*slice.as_ptr().cast() }
     }
 
+    #[inline]
     fn from_slice_mut(slice: &mut [Self::Value]) -> &mut Self {
         assert_eq!(slice.len(), Self::WIDTH);
-        slice.try_into().unwrap()
+        unsafe { &mut *slice.as_mut_ptr().cast() }
     }
 
-    fn from_fn<F>(f: F) -> Self
-    where
-        F: FnMut(usize) -> Self::Value,
-    {
+    #[inline]
+    fn from_fn<F: FnMut(usize) -> Self::Value>(f: F) -> Self {
```

Comment:
Nit: I think the original version was easier to read, and is certainly more idiomatic for very long type declarations like the one here.